pkg/reconciler/clusteringress/resources/virtual_service.go

```diff
@@ -155,6 +191,17 @@ func dedup(hosts []string) []string {
 	return sets.NewString(hosts...).List()
 }
 
+func retainLocals(hosts []string) []string {
```

Comment:
perhaps make this return strings.Set? Then you won't have to convert it to set for intersection later.

Reply:
I will follow up with this change, since it result in a lot of ordering changes in arrays, etc which result in a larger test change.